test/cpp/end2end/grpclb_end2end_test.cc

```diff
@@ -606,49 +612,28 @@ TEST_F(SingleBalancerTest, RepeatedServerlist) {
       kServerlistDelayMs);
 
   // Send num_backends/2 requests.
-  auto statuses_and_responses = SendRpc(kMessage_, num_backends_ / 2);
-  // only the first half of the backends will receive them.
+  CheckRpcSendOk(num_backends_ / 2);
+  // only half of the backends will receive them.
+  size_t num_requests = 0;
   for (size_t i = 0; i < backends_.size(); ++i) {
-    if (i < backends_.size() / 2)
-      EXPECT_EQ(1U, backend_servers_[i].service_->request_count())
-          << "for backend #" << i;
-    else
-      EXPECT_EQ(0U, backend_servers_[i].service_->request_count())
-          << "for backend #" << i;
-  }
-  EXPECT_EQ(statuses_and_responses.size(), num_backends_ / 2);
-  for (const auto& status_and_response : statuses_and_responses) {
-    const Status& status = status_and_response.first;
-    const EchoResponse& response = status_and_response.second;
-    EXPECT_TRUE(status.ok()) << "code=" << status.error_code()
-                             << " message=" << status.error_message();
-    EXPECT_EQ(response.message(), kMessage_);
+    num_requests += backend_servers_[i].service_->request_count();
```

Comment:
I'm not sure this check is actually telling us anything, since it doesn't say anything about the distribution of requests across backends.  And it can't possibly fail, because if we already sent N successful requests, then the total number of requests served by all backends must be N.

Stepping back a bit, what is the overall purpose of this test?  It seems like all we're really testing here is that if round_robin gets an update that is identical to its existing list of addresses, that it won't wind up changing anything.  Would it be better to just test that behavior in the round_robin test directly?

Reply:
I'm fine with adding that new test in another PR, but then let's just delete this test in this PR.  As far as I can tell, this test is not proving anything -- as currently written, it would still pass, even if the grpclb code didn't do anything special to handle duplicate serverlists.